protobuff-support/src/main/java/io/micronaut/protobuf/convert/ByteBufToProtoMessageConverter.java

```diff
@@ -56,10 +59,10 @@ public ByteBufToProtoMessageConverter(ProtobufferCodec codec) {
 
     private Optional<Message> rehydrate(ByteBuf object, Message.Builder builder) {
         try {
-            builder.mergeFrom(new ByteBufInputStream(object), codec.getExtensionRegistry());
+            builder.mergeFrom(new ByteBufInputStream(object.copy()), codec.getExtensionRegistry());
             return Optional.of(builder.build());
         } catch (IOException e) {
-            throw new IllegalStateException("Error parsing: " + e.getMessage());
+            return Optional.empty();
```

Comment:
Rather than just returning `Optional.empty()` I believe the correct think to do is call `context.reject(..)` with the exception. The `context` is passed to the `convert` method.

This is way if anything does care about the exception they can retrieve it and deal with it.

Reply:
We can use `public ByteBufInputStream(ByteBuf buffer, boolean releaseOnClose)` to avoid memory leak and explicit `.close()` call